packages/dynamic_layouts/lib/src/wrap_layout.dart

```diff
@@ -0,0 +1,271 @@
+      case Axis.vertical:
+        addedSize = currentSizeUsed + childSize.width + crossAxisSpacing;
+        break;
+      case Axis.horizontal:
+        addedSize = currentSizeUsed + childSize.height + mainAxisSpacing;
+        break;
+    }
+
+    if (addedSize > crossAxisExtent && model.last.currentSizeUsed > 0.0) {
+      switch (scrollDirection) {
+        case Axis.vertical:
+          model.add(
+            _RunMetrics(
+              maxSliver: childSize.height + mainAxisSpacing,
+              currentSizeUsed: childSize.width + crossAxisSpacing,
+              scrollOffset:
+                  scrollOffset + model.last.maxSliver + mainAxisSpacing,
+            ),
+          );
+          break;
+        case Axis.horizontal:
+          model.add(
+            _RunMetrics(
+              maxSliver: childSize.width + crossAxisSpacing,
+              currentSizeUsed: childSize.height + mainAxisSpacing,
+              scrollOffset:
+                  scrollOffset + model.last.maxSliver + crossAxisSpacing,
+            ),
+          );
+          break;
+      }
+
+      return DynamicSliverGridGeometry(
+        scrollOffset: model.last.scrollOffset,
+        crossAxisOffset: 0.0,
+        mainAxisExtent: childSize.height + mainAxisSpacing,
+        crossAxisExtent: childSize.width + crossAxisSpacing,
+      );
+    } else {
+      model.last.currentSizeUsed = addedSize;
+    }
+
+    switch (scrollDirection) {
+      case Axis.vertical:
+        if (childSize.height + mainAxisSpacing > model.last.maxSliver) {
+          model.last.maxSliver = childSize.height + mainAxisSpacing;
+        }
+        break;
+      case Axis.horizontal:
+        if (childSize.width + crossAxisSpacing > model.last.maxSliver) {
+          model.last.maxSliver = childSize.width + crossAxisSpacing;
+        }
+        break;
+    }
+
+    return DynamicSliverGridGeometry(
+      scrollOffset: scrollOffset,
+      crossAxisOffset: currentSizeUsed,
+      mainAxisExtent: childSize.height,
+      crossAxisExtent: childSize.width,
+    );
+  }
+
+  @override
+  bool reachedTargetScrollOffset(double targetOffset) {
+    if (model.last.scrollOffset > targetOffset) {
+      return true;
+    }
+    return false;
+  }
+}
+
+// TODO(snat-s): YourWrapDelegate extends SliverGridDelegate
+
+/// A [SliverGridDelegate] that makes it possible to create grids that wrap and
+/// have variably sized tiles.
+///
+/// For example, if the grid is vertical, this delegate will create a layout
+/// where the children are layed out until they fill the horizontal axis and then
+/// they pass to the second row. If the grid is horizontal, this delegate will
+/// do the same but it will fill the vertical axis and will pass to another
+/// column until it finishes.
+///
+/// This delegate creates grids with different sized tiles. But you
+/// can have equally sized tiles if you modify the `childCrossAxisExtent` and
+/// the `childMainAxisExtent` to be the same size.
```

Comment:
Just a suggestion... maybe

```suggestion
/// This delegate creates grids with different sized tiles. Tiles can
/// be have fixed dimensions if `childCrossAxisExtent` or
/// `childMainAxisExtent` are provided.
```